server/notification-providers/grafana-oncall.js

```diff
@@ -0,0 +1,41 @@
+const NotificationProvider = require("./notification-provider");
+const axios = require("axios");
+const { DOWN, UP } = require("../../src/util");
+
+class GrafanaOncall extends NotificationProvider {
+
+    name = "GrafanaOncall";
+
+    async send(notification, msg, monitorJSON = null, heartbeatJSON = null) {
+        let okMsg = "Sent Successfully.";
+        try {
+            if (heartbeatJSON["status"] === DOWN) {
```

Comment:
You are missing the testing button and certificate handling

```suggestion
            if (heartbeatJSON === null) {
                let grafanaupdata = {
                    title: "General notification",
                    message: msg,
                    state: "alerting",
                };
                await axios.post(
                    notification.GrafanaOncallURL,
                    grafanaupdata
                );
                return okMsg;
            if (heartbeatJSON["status"] === DOWN) {
```